packages/terralib/lua/TerraLib.lua

```diff
@@ -212,43 +244,72 @@ local function createLayer(name, dSetName, connInfo, type, addSpatialIdx, srid)
 			binding.te.da.DataSourceManager.getInstance():insert(ds)
 		end
 
-		if not ds:dataSetExists(dSetName) then
-			binding.te.da.DataSourceManager.getInstance():detach(dsId)
-			ds:close()
-			customError("It was not possible to find data set '"..dSetName.."' of type '"..type.."'. Layer '"..name.."' does not created.")
-		end
-
 		local dSetType
 		local dSet
 		local env = nil
 		local sridReal = 0
+		local id = binding.GetRandomicId()
+
+		if not (type == "WMS2") then
```

Comment:
In an `if`, when you know that you are going to use an `else`, never use `not` in the if. Use only positive conditions and leave the negative case to the `else`.